src/vivarium/framework/engine.py

```diff
@@ -202,16 +203,28 @@ def initialize_simulants(self):
         )
         self._clock.step_forward()
 
-    def step(self):
-        self._logger.debug(self._clock.time)
+    def step(self, log_step: bool = True):
+        if log_step:
+            self._logger.debug(self._clock.time)
+
         for event in self.time_step_events:
             self._lifecycle.set_state(event)
             self.time_step_emitters[event](self._population.get_population(True).index)
+
         self._clock.step_forward()
 
-    def run(self):
-        while self._clock.time < self._clock.stop_time:
-            self.step()
+    def take_steps(self, number_of_steps: int, progress_bar: bool = False) -> None:
+        log_step = not progress_bar
+        with tqdm.tqdm(total=number_of_steps, disable=not progress_bar) as pbar:
```

Comment:
Is it not possible for this logic to live inside the logging system? It feels more natural there.

Reply:
I think it is very not advisable.  The logger is a very thin wrapper around normal logging conventions.  Making good and informative progress bars requires a lot of context-specific state information that I think would be hard to generalize reasonably into the logging system.

On a more theoretical point, I think of progress bars as kind of an animation trick for your terminal that obscures the guarantees of a logging stream (a stream of discrete messages), which make me a bit hesitant to conflate the two.